pkg/asset/cluster/aws/aws.go

```diff
@@ -30,6 +33,16 @@ func Metadata(clusterID, infraID string, config *types.InstallConfig) *awstypes.
 // PreTerraform performs any infrastructure initialization which must
 // happen before Terraform creates the remaining infrastructure.
 func PreTerraform(ctx context.Context, clusterID string, installConfig *installconfig.InstallConfig) error {
+	if installConfig.Config.Publish == "External" && installConfig.Config.BaseDomain != "" && installConfig.Config.Platform.AWS.PublicZoneID != "" {
+		zone, err := awsconfig.GetPublicZoneByID(installConfig.Config.Platform.AWS.PublicZoneID)
+		if err != nil {
+			return err
+		}
+		if strings.ToLower(strings.TrimSuffix(aws.StringValue(zone.Name), ".")) != strings.ToLower(installConfig.Config.BaseDomain) {
+			return errors.New("the public zone id does not match the base domain")
+		}
+	}
```

Comment:
I think this should be part of the `pkg/assets/installconfig/aws/validations.go` , no?

Reply:
Removed.